File: src/clients/java/src/tigerbeetle-java/src/main/java/com/tigerbeetle/RequestException.java
package com.tigerbeetle;

public final class RequestException extends Exception {

    public interface Status {
        byte OK = 0;
        byte TOO_MUCH_DATA = 1;
        byte INVALID_OPERATION = 2;
        byte INVALID_DATA_SIZE = 3;
    }

    private final byte status;

    public RequestException(byte status) {
        this.status = status;
    }

    public byte getStatus() {
        return status;
    }

    @Override
    public String getMessage() {
        return toString();
    }

    @Override
    public String toString() {
        switch (status) {

            case Status.TOO_MUCH_DATA:
                return "Too much data provided on this batch.";

            case Status.INVALID_OPERATION:
                return "Invalid operation. Check if this client is compatible with the server's version.";

            case Status.INVALID_DATA_SIZE:
                return "Invalid data size. Check if this client is compatible with the server's version.";

            default:
                return "Unknown error status " + status;
        }
    }

}

File: src/clients/dotnet/TigerBeetle/RequestException.cs
using System;

namespace TigerBeetle;

public sealed class RequestException : Exception
{
    public PacketStatus Status { get; }

    internal RequestException(PacketStatus status)
    {
        Status = status;
    }

    public override string Message
    {
        get
        {
            switch (Status)
            {
                case PacketStatus.TooMuchData: return "Too much data provided on this batch.";
                case PacketStatus.InvalidOperation: return "Invalid operation.";
                case PacketStatus.InvalidDataSize: return "Invalid data size.";
                case PacketStatus.ClientEvicted: return "Client was evicted.";
                case PacketStatus.ClientReleaseTooLow: return "Client was evicted: release too old.";
                case PacketStatus.ClientReleaseTooHigh: return "Client was evicted: release too new.";
                case PacketStatus.ClientShutdown: return "Client was closed.";
                default: return "Unknown error status " + Status;
            }
        }
    }
}
